### PR TITLE
Change the joi options to a specific route.

### PR DESCRIPTION
**Pull request by carvalhoviniciusluiz**
Change the joi options to a specific route.

### e.g 
Config server
```javascript
  server.use(validator({
    allowUnknown: false
  }, {})
```
Config route
```javascript
  server.post({
    path: '/change_password',
    validation: {
      body: Joi.object().keys({
        password: Joi.string().required(),
        password_confirmation: Joi.any().valid(Joi.ref('password')).required()
      }).required()
    },
    joi: { allowUnknown: true } // <-- here
  },
  controller.change)
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ const defaultKeysToValidate = ['params', 'body', 'query', 'user', 'headers', 'tr
 
 const middleware = (joiOptions, options) => (req, res, next) => {
   const validation = req.route.validation
+  const joiOpts = req.route.joi
```
Comment (maxnachlinger) [inline index.js:18]: How about:
```javascript
const localJoiOptions = req.route.joi || joiOptions
// snip
const result = joi.validate(toValidate, validation, localJoiOptions)
```

**Review by maxnachlinger — CHANGES_REQUESTED** [review]
Made one small stylistic comment. Could you also please add a test proving that when both server and route level options are present, the route level options are used? Docs and examples should also be updated as well.

File: index.js (hunk 1 above)
Comment (carvalhoviniciusluiz) [reply index.js:18]: I think there's a validation missing

```javascript
const localJoiOptions = typeof req.route.joi === 'object' ? req.route.joi : joiOptions
// snip
const result = joi.validate(toValidate, validation, localJoiOptions)
```

Comment (maxnachlinger) [reply index.js:18]: This is a bit funny :) We could use `joi` to validate `req.route.joi` :)

**Comment by carvalhoviniciusluiz** [comment]
ok I will set up the tests and examples for use.

**Pull request closed without merging**